### ISSUE TITLE
M365 Learning pathways issue- CORS error

### ISSUE THREAD

**Issue by pankajpal89** [pnp/custom-learning-office-365#656]
## Describe the bug

Users are facing intermittent issue while accessing learning pathways site. Got the following error from console logs-

Get-started-with-Microsoft-Teams.aspx:1 Access to fetch at 'https://pnp.github.io/custom-learning-office-365/learningpathways/v4/manifest.json' from origin 'https://ajg0.sharepoint.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Checked with the network team and they said "Github.com and github.io has been explicitly whitelisted on our proxies".

### Learning Pathways version number

>4.3.1.0

### Additional context

>Users are facing intermittent issue while accessing learning pathways pages.
![Learning pathways issue](https://user-images.githubusercontent.com/98089934/163227614-fb5290d2-aaeb-4f27-b29b-ff052aef3a61.PNG)

**Comment by v-ashrirao** [comment]
@pankajpal89, thank you for reporting this.

Based on the logs you have shared we think network is blocking user to access
https://pnp.github.io/custom-learning-office-365/.

Request you to refer similar issues reported by users related to CORS policy.
https://github.com/pnp/custom-learning-office-365/issues/403
https://github.com/pnp/custom-learning-office-365/issues/196

In case more details are required please let us know.

**Comment by pankajpal89** [comment]
We have checked with our network team and they said "Github.com and github.io" has been explicitly whitelisted on our proxies".

**Comment by pankajpal89** [comment]
Checked with the network team and they said "Github.com and github.io has been explicitly whitelisted on our proxies".

**Comment by pankajpal89** [comment]
@v-ashrirao thank you for your response. 
We checked with the network team and they said "Github.com and github.io has been explicitly whitelisted on our proxies".

**Comment by dcashpeterson** [comment]
@pankajpal89 you need to make sure the full path is on the allow list. https://pnp.github.io/custom-learning-office-365/

**Comment by pankajpal89** [comment]
@dcashpeterson Network team has confirmed that above URL is whitelisted in proxy. Below is their response-

"the proxy whitelisting is done by Domain and Subdomain. We cannot put in full path specially “/”.and I can verify that pnp.github.io is whitelisted on the proxy"

**Comment by dcashpeterson** [comment]
Are you using any custom content packs? If so those repos need to be on the allow list as well. You should also check and see if https://support.office.com/ and https://docs.microsoft.com/ are on the allow list as well as that is where the content comes from.  Unfortunately, we cant really troubleshoot CORS related issues. If your team is saying that github is on the allow list as well as support and docs then you would need to do a trace with them to figure out what was blocking what.

**Comment by v-ashrirao** [comment]
@pankajpal89,
Is there any other information you need on this? If not, can we close the issue?

**Comment by pankajpal89** [comment]
Network team's comment after reviewing fiddler trace-

"We are not seeing anything being blocked as per the fiddler trace provided. The only thing that is creating an issue is timeout on ntp.msn.com which is showing as 404 page not found. 

All the links for SharePoint are whitelisted and has Http Code 200 which means they are able to connect without any issue. "

**Comment by brightmatt** [comment]
We are experiencing the same problem but the odd part is, we have 3 users in the "owners" group, and 2 of them get the error and I do not.  We have tired both Chrome and Edge, are on the same network, have the same content filters in place, and do not see anything getting blocked.  Any help would be much appreciated.

**Comment by dcashpeterson** [comment]
@brightmatt have you looked at the responses previously posted? 

Based on the logs you have shared we think network is blocking user to access
https://pnp.github.io/custom-learning-office-365/.

Request you to refer similar issues reported by users related to CORS policy.
https://github.com/pnp/custom-learning-office-365/issues/403
https://github.com/pnp/custom-learning-office-365/issues/196

Can you paste the error you are getting from the console in the issue to confirm.

**Comment by Acscpr** [comment]
Também estou com o mesmo problema em meu site de Sharepoint. Acredito não ser porta de rede, porque eu testei em usuários diferentes dentro da rede interna da empresa e estando em home office e o erro permanece. Veja print em anexo.

Testes realizados:
- Usuários com e sem acesso administrativo ao site. 
Resultados: o user com acesso a admin continua acessando, e os demais não acessam. Mesmo liberando acesso de admin ao usuário com erro, não funciona!

- Exclusão e adição do app no site. 
Resultado: sem sucesso.

- Implementação do recurso em uma nova página
-Resultado: sem sucesso.

Peço ajuda para solucionar o problema. 

![MicrosoftTeams-image (9)](https://user-images.githubusercontent.com/68610887/170784961-1cac2240-cbfa-4a1e-9b80-01e21d69311b.png)
.

**Comment by dcashpeterson** [comment]
@Acscpr can you please create a new issue and reference this one. It gets challenging to manage issues if they have more than one sounds in it.
